chains/solana/contracts/programs/ccip-router/src/instructions/v1/messages.rs

```diff
@@ -56,3 +64,230 @@ pub struct LockOrBurnOutV1 {
 pub struct ReleaseOrMintOutV1 {
     pub destination_amount: u64, // TODO: u256 on EVM?
 }
+
+///////////////////
+// Ramp Messages //
+///////////////////
+
+const U160_MAX: U256 = U256::from_words(u32::MAX as u128, u128::MAX);
+
+pub fn validate_solana2any(
+    msg: &Solana2AnyMessage,
+    dest_chain: &DestChain,
+    token_config: &BillingTokenConfig,
+) -> Result<()> {
+    require!(
+        dest_chain.config.is_enabled,
+        CcipRouterError::DestinationChainDisabled
+    );
+
+    require!(token_config.enabled, CcipRouterError::FeeTokenDisabled);
+
+    require_gte!(
+        dest_chain.config.max_data_bytes,
+        msg.data.len() as u32,
+        CcipRouterError::MessageTooLarge
+    );
+
+    require_gte!(
+        dest_chain.config.max_number_of_tokens_per_msg as usize,
+        msg.token_amounts.len(),
+        CcipRouterError::UnsupportedNumberOfTokens
+    );
+
+    validate_dest_family_address(msg, dest_chain.config.chain_family_selector)
+}
+
+fn validate_dest_family_address(
+    msg: &Solana2AnyMessage,
+    chain_family_selector: [u8; 4],
+) -> Result<()> {
+    const PRECOMPILE_SPACE: u32 = 1024;
+
+    let selector = u32::from_be_bytes(chain_family_selector);
+    // Only EVM is supported as a destination family.
+    require_eq!(
+        selector,
+        CHAIN_FAMILY_SELECTOR_EVM,
+        CcipRouterError::UnsupportedChainFamilySelector
+    );
+
+    require_eq!(msg.receiver.len(), 32, CcipRouterError::InvalidEVMAddress);
+
+    let address: U256 = U256::from_be_bytes(
+        msg.receiver
+            .clone()
+            .try_into()
+            .map_err(|_| CcipRouterError::InvalidEncoding)?,
+    );
+
+    require!(address <= U160_MAX, CcipRouterError::InvalidEVMAddress);
+
+    if let Ok(small_address) = TryInto::<u32>::try_into(address) {
+        require_gte!(
+            small_address,
+            PRECOMPILE_SPACE,
+            CcipRouterError::InvalidEVMAddress
+        )
+    };
+
+    Ok(())
+}
+
+#[cfg(test)]
+pub(crate) mod tests {
```

Comment:
Nit: is `(crate)` an artifact of automated refactoring? (asking because it tends to be).

Reply:
It was already there. There are a couple of test data factories that are reused in other tests, so it has to be public within some context. It doesn't make any difference whether that's restricted to the crate or not though, as this already has the `#[cfg(test)]` anyway, so I've changed this to a regular `pub` :+1: